usage/src/org/apache/cloudstack/quota/QuotaManagerImpl.java

```diff
@@ -0,0 +1,465 @@
+                    Date endDate = quotalistforaccount.get(0).getEndDate();
+                    BigDecimal aggrUsage = new BigDecimal(0);
+                    for (QuotaUsageVO entry : quotalistforaccount) {
+                        if (startDate.compareTo(entry.getStartDate()) != 0) {
+                            QuotaBalanceVO lastrealbalanceentry = _quotaBalanceDao.findLastBalanceEntry(account.getAccountId(), account.getDomainId(), startDate);
+                            Date lastbalancedate;
+                            if (lastrealbalanceentry != null) {
+                                lastbalancedate = lastrealbalanceentry.getUpdatedOn();
+                                aggrUsage = aggrUsage.add(lastrealbalanceentry.getCreditBalance());
+                            } else {
+                                lastbalancedate = new Date(0);
+                            }
+
+                            List<QuotaBalanceVO> creditsrcvd = _quotaBalanceDao.findCreditBalance(account.getAccountId(), account.getDomainId(), lastbalancedate, endDate);
+                            for (QuotaBalanceVO credit : creditsrcvd) {
+                                aggrUsage = aggrUsage.add(credit.getCreditBalance());
+                            }
+
+                            QuotaBalanceVO newbalance = new QuotaBalanceVO(account.getAccountId(), account.getDomainId(), aggrUsage, endDate);
+                            // s_logger.info("Balance entry=" + aggrUsage + " on Date=" + endDate);
+                            _quotaBalanceDao.persist(newbalance);
+                            aggrUsage = new BigDecimal(0);
+                        }
+                        startDate = entry.getStartDate();
+                        endDate = entry.getEndDate();
+                        aggrUsage = aggrUsage.subtract(entry.getQuotaUsed());
+                    }
+                    // update is quota_accounts
+                    QuotaAccountVO quota_account = _quotaAcc.findById(account.getAccountId());
+                    s_logger.info("Updating quota account bal=" + aggrUsage + " date=" + endDate);
+                    if (quota_account == null) {
+                        quota_account = new QuotaAccountVO(account.getAccountId());
+                        quota_account.setQuotaBalance(aggrUsage);
+                        quota_account.setQuotaBalanceDate(endDate);
+                        _quotaAcc.persist(quota_account);
+                    } else {
+                        quota_account.setQuotaBalance(aggrUsage);
+                        quota_account.setQuotaBalanceDate(endDate);
+                        _quotaAcc.update(account.getAccountId(), quota_account);
+                    }
+                }// balance processed
+            } // END ACCOUNT
+            jobResult = true;
+        } catch (Exception e) {
+            s_logger.error("Quota Manager error", e);
+            e.printStackTrace();
```

Comment:
Printing the stack trace goes to console -- incurring overhead and likely being ignored.  It should be removed.